front_end/models/react_native/FuseboxBindingsModel.ts

```diff
@@ -0,0 +1,109 @@
+    void this.enable(target);
+  }
+
+  private bindingCalled(event: BindingCalledEventTargetEvent): void {
+    const serializedMessage = event.data.payload;
+
+    try {
+      const {domain, message} = JSON.parse(serializedMessage);
+      this.dispatchMessageToDomainEventListeners(domain, message);
+    } catch (err) {
+      throw new Error('Failed to parse bindingCalled event payload:', err);
+    }
+  }
+
+  subscribeToDomainMessages(domainName: DomainName, listener: DomainMessageListener): void {
+    let listeners = this.domainToListeners.get(domainName);
+    if (!listeners) {
+        listeners = new Set();
+        this.domainToListeners.set(domainName, listeners);
+    }
+
+    listeners.add(listener);
+  }
+
+  unsubscribeFromDomainMessages(domainName: DomainName, listener: DomainMessageListener): void {
+    const listeners = this.domainToListeners.get(domainName);
+    if (!listeners) {
+      throw new Error(`Failed to unsubscribe from domain ${domainName}: no listeners were previously attached`);
+    }
+
+    listeners.delete(listener);
+  }
+
+  private dispatchMessageToDomainEventListeners(domainName: DomainName, message: JSONValue): void {
+    const listeners = this.domainToListeners.get(domainName);
+    if (!listeners) {
+      // No subscriptions, no need to throw, just don't notify.
+      return;
+    }
+
+    for (const listener of listeners) {
+      listener(message);
+    }
+  }
+
+  async initializeDomain(domainName: DomainName): Promise<void> {
+    const runtimeModel = this.target().model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      throw new Error(`Failed to initialize domain ${domainName}: runtime model is not available`);
+    }
+
+    await runtimeModel.agent.invoke_evaluate({expression: `void ${RUNTIME_GLOBAL}.initializeDomain('${domainName}')`});
+  }
+
+  async sendMessage(domainName: DomainName, message: JSONValue): Promise<void> {
+    const runtimeModel = this.target().model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      throw new Error(`Failed to send message for domain ${domainName}: runtime model is not available`);
+    }
+
+    const serializedMessage = JSON.stringify(message);
+
+    await runtimeModel.agent.invoke_evaluate({expression: `${RUNTIME_GLOBAL}.sendMessage('${domainName}', '${serializedMessage}')`});
+  }
+
+  async enable(target: SDK.Target.Target): Promise<void> {
+    const runtimeModel = target.model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      throw new Error('Failed to enable FuseboxBindings model: runtime model is not available');
+    }
+
+    await runtimeModel.agent.invoke_evaluate({expression: `${RUNTIME_GLOBAL}.BINDING_NAME`})
+      .then(response => response.result.value)
+      .then(bindingName => runtimeModel.agent.invoke_addBinding({name: bindingName}));
+  }
+}
+
+SDK.SDKModel.SDKModel.register(FuseboxBindingsModel, {capabilities: SDK.Target.Capability.JS, autostart: true});
```

Comment:
Should consider gating this with the flag that I've recently added for React DevTools

Reply:
Resolved: changed `autostart` to false, so React DevTools panel will be responsible for starting this model